### PR TITLE
`fn render` will fail if the function config has multiple resources.

### LINKED ISSUE #1959: Fn render should error when there are multiple resources in a fn config file
Current behavior: Only the first one is respected and the rest are ignored.

### PR DESCRIPTION
**Pull request by Shell32-Natsu**
close #1959

### CODE REVIEW

File: thirdparty/kyaml/runfn/runfn.go [hunk 1]
```diff
@@ -386,23 +386,24 @@ func getUIDGID(asCurrentUser bool, currentUser currentUserFunc) (string, error)
 	return fmt.Sprintf("%s:%s", u.Uid, u.Gid), nil
 }
 
-// etFunctionConfig returns yaml representation of functionConfig that can
+// getFunctionConfig returns yaml representation of functionConfig that can
 // be provided to a function as input.
 func (r *RunFns) getFunctionConfig() (*yaml.RNode, error) {
```
Comment (natasha41575) [inline thirdparty/kyaml/runfn/runfn.go:391]: can this function be moved to fnruntime/runner.go and shared between `eval` and `render`? What's the reason for keeping them separate?

**Comment by natasha41575** [comment]
also please keep an eye on https://github.com/GoogleContainerTools/kpt/pull/2133, I think there is some overlap and may affect where you want to do this check. I guess it depends on which one we want to merge first

**Comment by Shell32-Natsu** [comment]
> also please keep an eye on #2133, I think there is some overlap and may affect where you want to do this check. I guess it depends on which one we want to merge first

I think it's probably better to check this during validating. One of my concern is we will need to read the config file twice, one for validating one for using. @droot WDYT?

**Comment by Shell32-Natsu** [comment]
Another problem is `fn eval` doesn't validate the content in function config file. We should probably keep consistent with `render`.

**Comment by natasha41575** [comment]
@Shell32-Natsu I just updated #2133 to share a function to validate the function config file. Please take a look. I think it should also cover #1959.

**Comment by Shell32-Natsu** [comment]
> @Shell32-Natsu I just updated #2133 to share a function to validate the function config file. Please take a look. I think it should also cover #1959.

Right, the method in #2133 is better. Close this one.

**Pull request closed without merging**